hazelcast/src/main/java/com/hazelcast/config/Config.java

```diff
@@ -530,7 +530,10 @@ public String getProperty(String name) {
      * Hazelcast System Properties</a>
      */
     public Config setProperty(String name, String value) {
-        properties.put(name, value);
+        if (!isNullOrEmptyAfterTrim(name) && !isNullOrEmptyAfterTrim(value)) {
```

Comment:
Since we don't want to let the user setting the property if either the `key` or the `value` is `null`, maybe throwing an `IllegalArgumentException` would be more appropriate than silently ignoring the call. WDYT?

Reply:
fixed.
throw IllegalArgumentExeption
and added two tests.